src/docfx/lib/log/Telemetry.cs

```diff
@@ -34,10 +35,19 @@ internal static class Telemetry
 
         private static string s_correlationId = EnvironmentVariable.CorrelationId ?? Guid.NewGuid().ToString("N");
 
+        static Telemetry()
+        {
+            s_telemetryClient.Context.GlobalProperties["OS"] = RuntimeInformation.OSDescription ?? "<null>";
+            s_telemetryClient.Context.GlobalProperties["Version"] = typeof(Telemetry).Assembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>()?.InformationalVersion ?? "<null>";
+            s_telemetryClient.Context.GlobalProperties["CorrelationId"] = EnvironmentVariable.CorrelationId ?? Guid.NewGuid().ToString("N");
```

Comment:
Ok, if this to be put into `GlobalProperties`, ensure it is using the same value:
```suggestion
            s_telemetryClient.Context.GlobalProperties["CorrelationId"] = s_correlationId;
```

Consider doing this for other global properties